### ISSUE TITLE
Breaking changes in the 0.8 series - upgrade to 0.9

### ISSUE THREAD

**Issue by toranb** [toranb/ember-cli-simple-store#5]
As of 0.8 the public api for findById and findOne are now computed (previously they wouldn't update when something new was pushed into the store)

This should be considered a breaking change but a necessary one - I'll be working through the upgrade of a few client apps tonight to see how this causes pain/ or not. The big reason for the change is that if you return from a route (not waiting for an xhr to finish) the databound Array would update ... but a child route that looked up the same element by id would not (inconsistent / incorrect depending on your view of the world).

If you run up against this pain point today stick w/ 0.7.1 as it was still sync lookup based. Also report issues here so I can help work through them :)

I'll provide some updated examples of how to use this as we approach the 1.0

**Comment by toranb** [comment]
Quick update / apology to everyone who pulled in the 0.8 series of this project. After much debate/discussion it's obvious the changes to findById and findOne are breaking (by returning ObjectProxy instead of null/the model directly). This should have been communicated / published in a beta like npm release instead. Before the 1.0 I didn't want to introduce any breaking api changes so I'll cut a release of 0.9 tonight that takes the project back to the non computed behavior (shown in 0.7.x)

**Comment by toranb** [comment]
The 0.9.0 release is out on npm and solves the issues w/ computed findById/findOne that got released in the 0.8.x series of this project

**Comment by toranb** [comment]
After talking with a few people in the community another valid reason for this change is consistency. Today the array based find methods return a computed value that will be updated when anything is pushed into the store. But if you findById you don't get this same experience leaving people confused about how it works/why it works/etc

The plan for 1.0 is to verify this assumption/ talk about the migration path forward using ObjectProxy

**Comment by toranb** [comment]
Update: I've pushed a beta release w/ findById computed again if you really want that :)

"ember-cli-simple-store": "1.0.0-beta.1"

This includes an update that 0.8.5 didn't have - your objects functions are first class so instead of

``` js
this.get("model").get("content").foobar();
```

you can do this instead

``` js
this.get("model").foobar();
```

**Comment by toranb** [comment]
We did 2 other releases this week.

"ember-cli-simple-store": "1.0.0-beta.2" 
"ember-cli-simple-store": "1.0.0-beta.3" 

The latest release shown above includes findOne computed (along w/ the isDirty fix that went into 0.9.1)
